workbench_utils.py

```diff
@@ -2772,8 +2773,30 @@ def download_remote_file(config, url):
         print('Error: ' + message)
 
     # create_media() references the path of the downloaded file.
-    downloaded_file_path = os.path.join(config['input_dir'], url.split("/")[-1])
-    open(downloaded_file_path, 'wb+').write(response.content)
+    subdir = config['input_dir'] + '/' + re.sub('[^A-Za-z0-9]+', '_', node_csv_row[config['id_field']])
+    Path(subdir).mkdir(parents=True, exist_ok=True)
+
+    if config["use_node_title_for_media"]:
+        filename = re.sub('[^A-Za-z0-9]+', '_', node_csv_row['title'])
+        if filename[-1] == '_':
+            filename = filename[:-1]
+        downloaded_file_path = os.path.join(subdir, filename)
+    else:
+        downloaded_file_path = os.path.join(subdir, url.split("/")[-1])
+
+    file_extension = os.path.splitext(url)[1]
+    f = open(downloaded_file_path, 'wb+')
+    f.write(response.content)
+    f.close
+    mime = magic.from_file(downloaded_file_path, mime=True)
+    ext = mimetypes.guess_extension(mime)
+    if ext == '.jpe':
+        ext = '.jpg'
+    if file_extension =='':
+        os.rename(downloaded_file_path, downloaded_file_path + ext)
+        downloaded_file_path = downloaded_file_path + ext
+
```

Comment:
The "download" version of the file remains in the id-based subdirectory:

```
01
├── edcartoons_842.jpg
└── Motor_is_in_the_rear
02
├── edcartoons_469.jpg
└── Losing_Interest
03
├── Bowen_Islandora_Inn.jpg
└── view.jpg
04
├── download.jpg
└── Nelson_Caf.jpg
```
We probably want to not leave two copies of the file on disk, as they would chew up space twice as fast. Can we delete the original copy after it's renamed?

Reply:
Giving it a scour now -

Reply:
Should be good now - I hadn't allowed for selecting media title on URLs with extensions.  This should get rid of double save files as well as the 400 errors (which were caused by trying to ingest a file without an approved extension.